org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/configuration/GradleArguments.java

```diff
@@ -93,7 +98,17 @@ private String toNonEmpty(List<String> stringValues, String defaultMessage) {
     public void applyTo(GradleConnector connector) {
         connector.forProjectDirectory(this.rootDir);
         connector.useGradleUserHomeDir(this.gradleUserHome);
-        this.gradleDistribution.apply(connector);
+        if (this.gradleDistribution instanceof LocalGradleDistribution) {
```

Comment:
❌ `GradleDistribution` should have `applyTo(Connector)`

Reply:
Isn't it a problem if we expose those methods? I aimed for a minimal API.

Reply:
Fixed